### PR TITLE
Rep 2357: Generate and add a x-trans-id to feed invalidation requests

### PR DESCRIPTION
**Pull request by mknocke**
A x-trans-id is logged and added as a header for each invalidation request.

### CODE REVIEW

**Comment by adrianjgeorge** [comment]
ok to test

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/FeedCacheInvalidator.java [hunk 1]
```diff
@@ -77,15 +78,20 @@ public void setFeeds(List<AuthFeedReader> feeds) {
 
     @Override
     public void run() {
+
+        // Generate trans-id here so it is the same between multiple pages
+        String traceID = UUID.randomUUID().toString();
         while (!done) {
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/FeedCacheInvalidator.java]: you didn't add it to the slf4j context, without that it won't show up in our logs

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/FeedCacheInvalidator.java [hunk 2]
```diff
@@ -77,15 +78,20 @@ public void setFeeds(List<AuthFeedReader> feeds) {
...
 
-            List<String> userKeys = new ArrayList<String>();
-            List<String> tokenKeys = new ArrayList<String>();
+            LOG.debug("Feed invalidation request for trans-id: " + traceID);
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/FeedCacheInvalidator.java]: you don't need to explicitly add the trace id, if you put it in the context it'll get there automatically.

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/sax/SaxAuthFeedReader.java [hunk 3]
```diff
@@ -112,11 +109,12 @@ public CacheKeys getCacheKeys() throws FeedException {
...
     }
 
-    private ServiceClientResponse getFeed() throws FeedException {
+    private ServiceClientResponse getFeed(String traceID) throws FeedException {
 
         ServiceClientResponse resp;
-        final Map<String, String> headers = new HashMap<String, String>();
+        final Map<String, String> headers = new HashMap<>();
 
+        headers.put(CommonHttpHeader.TRACE_GUID.toString(), traceID);
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/sax/SaxAuthFeedReader.java]: you didn't do the check to see if it needed to be added

**Comment by adrianjgeorge** [comment]
you don't have any checks making sure it's getting into the logs nor that it's actually on the requests

**Comment by dmnjohns** [comment]
retest this please

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/sax/SaxAuthFeedReader.java [hunk 4]
```diff
@@ -112,17 +108,18 @@ public CacheKeys getCacheKeys() throws FeedException {
...
     }
 
-    private ServiceClientResponse getFeed() throws FeedException {
+    private ServiceClientResponse getFeed(String traceID) throws FeedException {
 
         ServiceClientResponse resp;
-        final Map<String, String> headers = new HashMap<String, String>();
+        final Map<String, String> headers = new HashMap<>();
 
+        /* TODO: check if necessary to add trans id header */
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/sax/SaxAuthFeedReader.java]: got a todo here, that you should probably take care of

File: repose-aggregator/components/filters/client-auth/src/test/java/org/openrepose/filters/clientauth/atomfeed/FeedCacheInvalidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    private ListAppender app;
+
+    @Before
+    public void setUp() {
+        LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+        app = ((ListAppender) (ctx.getConfiguration().getAppender("List0"))).clear();
+        datastore = mock(Datastore.class);
+    }
+
+    @Test
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/test/java/org/openrepose/filters/clientauth/atomfeed/FeedCacheInvalidatorTest.java]: what happened to making sure it was applying a guid?

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/sax/SaxAuthFeedReader.java [hunk 6]
```diff
@@ -66,17 +63,23 @@
     private AdminTokenProvider provider;
 
     private AkkaServiceClient akkaServiceClient;
+    private boolean isOutboundTracing = false;
 
-    public SaxAuthFeedReader(ServiceClient client, AkkaServiceClient akkaClient, String feedHead, String feedId) {
+    public SaxAuthFeedReader(ServiceClient client, AkkaServiceClient akkaClient, String feedHead, String feedId, boolean systemModel) {
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/sax/SaxAuthFeedReader.java]: parameter name

**Comment by adrianjgeorge** [comment]
fix the name and i'll merge it

**Pull request merged**